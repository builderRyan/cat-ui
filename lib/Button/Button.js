"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
// import * as PropTypes from 'prop-types';
var classnames_1 = require("classnames");
var react_1 = require("react");
/**
 * 按钮
 */
var Button = /** @class */ (function (_super) {
    __extends(Button, _super);
    function Button() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Button.prototype.render = function () {
        var _a;
        var _b = this.props, prefix = _b.prefix, className = _b.className, theme = _b.theme;
        var classes = classnames_1.default(prefix + "-button", (_a = {},
            _a[prefix + "-button-" + theme] = theme,
            _a), className);
        return React.createElement("div", null, "Button");
    };
    Button.defaultProps = {
        prefix: 'hmly',
        theme: 'primary',
        size: 'md',
        disabled: false
    };
    return Button;
}(react_1.Component));
exports.default = Button;
//# sourceMappingURL=Button.js.map