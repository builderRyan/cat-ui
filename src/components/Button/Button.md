Example

```jsx
<Button prop1='str' prop2={22} prop3='option1' />
```
