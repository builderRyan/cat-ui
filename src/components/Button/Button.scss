@import '../../style/index.scss';

.#{$prefix}-button {
  display: inline-block;
  border-radius: 2px;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.2px;
  cursor: pointer;
  outline: 0;
  border: 1px solid transparent;
  &:active {
    outline: none;
    box-shadow: none;
    text-decoration: none;
  }

  &.#{$prefix}-button-sm {
    padding: 6px 20px;
  }
  &.#{$prefix}-button-md {
    padding: 10px 25px;
  }
  &.#{$prefix}-button-lg {
    padding: 14px 29px;
    font-size: 16px;
    font-weight: bold;
  }

  &.#{$prefix}-button-primary {
    background-color: $primary-btn-color;
    color: #ffffff;
  }
  &.#{$prefix}-button-yellow {
    background-color: $yellow-btn-color;
    color: #ffffff;
  }
  &.#{$prefix}-button-white-primary {
    background-color: #ffffff;
    color: #9b9b9b;
    border: 1px solid rgba(83, 118, 130, 0.8);
  }
  &.#{$prefix}-button-white-cyan {
    background-color: #ffffff;
    color: $primary-btn-color;
    border: 1px solid $primary-btn-color;
  }

  &.#{$prefix}-button-block {
    width: 100%;
  }

  &.#{$prefix}-button-disabled {
    background-color: #a6a6a6;
    color: #ffffff;
    cursor: not-allowed;
    border: 1px solid transparent;
  }
}
