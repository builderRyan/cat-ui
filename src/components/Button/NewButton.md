Example

```jsx
<h3>size</h3>
<NewButton size='sm'>sm</NewButton>
<NewButton size='md'>md</NewButton>
<NewButton size='lg'>lg</NewButton>

<h3>theme</h3>
<NewButton theme='primary'>primary</NewButton>
<NewButton theme='yellow'>yellow</NewButton>
<NewButton theme='white-primary'>white-primary</NewButton>
<NewButton theme='white-cyan'>white-cyan</NewButton>

<h3>block</h3>
<div className='block-box'>
  <NewButton block>block</NewButton>
</div>
<div className='block-box'>
  <NewButton block theme='yellow'>block</NewButton>
</div>
<div className='block-box'>
  <NewButton block theme='white-primary'>block</NewButton>
</div>

<h3>disabled</h3>
<NewButton disabled>disabled</NewButton>
<NewButton disabled theme='yellow'>disabled</NewButton>
<NewButton disabled theme='white-primary'>disabled</NewButton>

<style>
  {`
    .hmly-button + .hmly-button {
      margin-left: 20px;
    }
    .block-box {
      width: 300px;
    }
    .block-box + .block-box {
      margin-top: 20px;
    }
  `}
</style>
```
