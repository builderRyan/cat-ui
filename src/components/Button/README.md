# Button 按钮

## Example

```jsx
<Button>Button</Button>  
```

## API

|   属性    |   类型    |  默认值   |                                         描述                                         |
|:---------:|:---------:|:---------:|:------------------------------------------------------------------------------------:|
|  preifx   | `string`  |   `cat`   |                                         前缀                                         |
| className | `string`  |     -     |                                         类名                                         |
|   style   | `object`  |     -     |                                       行内样式                                       |
|   theme   | `string`  | `primary` | 主题，可选 `primary` `primary-up` `yellow` `white-primary` `white-cyan` `borderless` |
| disabled  | `boolean` |   false   |                                       是否禁用                                       |
| htmlType  | `string`  |     -     |                                         类型                                         |
|   block   | `boolean` |   false   |                             是否将按钮宽度调整为其父宽度                             |
|   size    | `string`  |   `lg`    |                              尺寸，可选 `sm` `md` `lg`                               |
|  loading  | `boolean` |   false   |                                   设置按钮载入状态                                   |
|   icon    | `string`  |     -     |                                  设置按钮的图标类型                                  |
