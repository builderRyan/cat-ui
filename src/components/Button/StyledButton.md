Example

```jsx
<h3>size</h3>
<StyledButton size='sm'>sm</StyledButton>
<StyledButton size='md'>md</StyledButton>
<StyledButton size='lg'>lg</StyledButton>

<h3>theme</h3>
<StyledButton theme='primary'>primary</StyledButton>
<StyledButton theme='yellow'>yellow</StyledButton>
<StyledButton theme='white-primary'>white-primary</StyledButton>
<StyledButton theme='white-cyan'>white-cyan</StyledButton>

<h3>block</h3>
<div className='block-box'>
  <StyledButton block>block</StyledButton>
</div>
<div className='block-box'>
  <StyledButton block theme='yellow'>block</StyledButton>
</div>
<div className='block-box'>
  <StyledButton block theme='white-primary'>block</StyledButton>
</div>

<h3>disabled</h3>
<StyledButton disabled>disabled</StyledButton>
<StyledButton disabled theme='yellow'>disabled</StyledButton>
<StyledButton disabled theme='white-primary'>disabled</StyledButton>

<style>
  {`
    .hmly-button + .hmly-button {
      margin-left: 20px;
    }
    .block-box {
      width: 300px;
    }
    .block-box + .block-box {
      margin-top: 20px;
    }
  `}
</style>
```
