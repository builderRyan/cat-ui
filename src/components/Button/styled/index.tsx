import styled from 'styled-components'
import { primaryBtnColor, yellowBtnColor } from '../../../styles/theme'

const StyledButton = styled.button`
  display: inline-block;
  border-radius: 4px;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.2px;
  cursor: pointer;
  outline: 0;
  border: 1px solid transparent;

  &:active {
    outline: none;
    box-shadow: none;
    text-decoration: none;
  }

  &.hmly-button-sm {
    padding: 5px 12px;
    font-size: 14px;
    font-weight: normal;
  }
  &.hmly-button-md {
    padding: 9px 24px;
    font-size: 14px;
    font-weight: 600;
  }
  &.hmly-button-lg {
    padding: 12px 36px;
    font-size: 16px;
    font-weight: bold;
  }

  &.hmly-button-primary {
    background-color: ${primaryBtnColor};
    color: #ffffff;
    &:hover {
      box-shadow: 0 6px 16px 0 rgba(255, 68, 68, 0.5);
      background-color: #ff6464;
    }
  }
  &.hmly-button-yellow {
    background-color: ${yellowBtnColor};
    color: #ffffff;
  }
  &.hmly-button-white-primary {
    background-color: #ffffff;
    color: #9b9b9b;
    border: 1px solid rgba(83, 118, 130, 0.8);
  }
  &.hmly-button-white-cyan {
    background-color: #ffffff;
    color: ${primaryBtnColor};
    border: 1px solid ${primaryBtnColor};
  }
  &.hmly-button-borderless {
    color: ${primaryBtnColor};
  }

  &.hmly-button-block {
    width: 100%;
  }

  &.hmly-button-disabled {
    background-color: #a6a6a6;
    color: #ffffff;
    cursor: not-allowed;
    border: 1px solid transparent;
  }
`;

export { StyledButton }
