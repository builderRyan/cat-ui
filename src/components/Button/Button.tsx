import * as React from 'react'
import { PureComponent } from 'react'
// import * as PropTypes from 'prop-types'
import classNames from 'classnames'
import { StyledButton } from './styled'

export interface ButtonProps {
  /** 前缀 */
  prefix?: string;
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** 主题 */
  theme?: 'primary' | 'yellow' | 'white-primary' | 'white-cyan';
  /** 是否禁用 */
  disabled?: boolean;
  /** 样式 */
  htmlType?: 'submit' | 'button' | 'reset';
  /** 是否将按钮宽度调整为其父宽度 */
  block?: boolean;
  /** 尺寸 */
  size?: 'sm' | 'md' | 'lg';
  /** 设置按钮的图标类型 */
  icon?: string;
  /** 子元素 */
  children?: React.ReactNode;
  /** 点击按钮时的回调 */
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

/**
 * 按钮
 */
class Button extends PureComponent<ButtonProps, any> {
  static defaultProps = {
    prefix: 'hmly',
    htmlType: 'button',
    theme: 'primary',
    size: 'md',
    disabled: false
  }

  handleClick: React.MouseEventHandler<HTMLButtonElement> = (event) => {
    const { disabled, onClick } = this.props
    console.log('Button:handleClick: disabled', disabled)
    if (disabled) return
    onClick && onClick(event)
  }

  render() {
    const { prefix, className, style, theme, size, block, disabled,
      htmlType, children, ...others } = this.props
    const classes = classNames(`${prefix}-button`, {
      [`${prefix}-button-${theme}`]: theme,
      [`${prefix}-button-${size}`]: size,
      [`${prefix}-button-block`]: block,
      [`${prefix}-button-disabled`]: disabled
    }, className)

    return (<StyledButton
      className={classes}
      style={style}
      type={htmlType}
      onClick={this.handleClick}
      {...others}>
      {children}
    </StyledButton>)
  }
}

export default Button
