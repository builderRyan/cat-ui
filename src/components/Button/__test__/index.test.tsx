import * as React from 'react';
import * as Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<Button>Button</Button>);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders props:style', () => {
    const wrapper = shallow(<Button style={{ fontSize: 24 }}>Button</Button>);
    const style = wrapper.find('button').props().style;
    expect(style).toEqual({ fontSize: 24 });
  });
  test('renders props:loading', () => {
    const wrapper = shallow(<Button loading>Button</Button>);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders props:loading,type', () => {
    const wrapper = shallow(
      <Button loading type='primary'>
        Button
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
