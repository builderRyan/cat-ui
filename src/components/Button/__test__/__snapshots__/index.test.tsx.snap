// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="cat-button cat-button--default cat-button--square cat-button--md cat-button--padding-md"
  disabled={false}
  style={Object {}}
  type="button"
>
  <span
    className="cat-button__children"
  >
    Button
  </span>
</button>
`;

exports[`Button renders props:loading 1`] = `
<button
  className="cat-button cat-button--default cat-button--square cat-button--md cat-button--loading cat-button--padding-md"
  disabled={false}
  style={Object {}}
  type="button"
>
  <Loading
    className="cat-button__loading"
    color="#537682"
    size={14}
    visible={true}
  />
  <span
    className="cat-button__children"
  >
    Button
  </span>
</button>
`;

exports[`Button renders props:loading,type 1`] = `
<button
  className="cat-button cat-button--primary cat-button--square cat-button--md cat-button--loading cat-button--padding-md"
  disabled={false}
  style={Object {}}
  type="button"
>
  <Loading
    className="cat-button__loading"
    color="#fff"
    size={14}
    visible={true}
  />
  <span
    className="cat-button__children"
  >
    Button
  </span>
</button>
`;
